Add `GITHUB_COM_TOKEN` information for manual testing

pjbgf opened the pull request:
For reference, the token is redacted as per debug logs below:
```
DEBUG: Env config
       'config': {
         'hostRules': [
           {'hostType': 'github', 'matchHost': 'github.com', 'token': '***********'}
         ]
       }
```

macedogm commented on Makefile:
```diff
@@ -3,7 +3,7 @@ RENOVATE_IMG := renovate/renovate:slim
 VALIDATE_FILE := docker run --rm -v $(shell pwd):/repo:ro -e LOG_LEVEL=debug \
 		$(RENOVATE_IMG) renovate-config-validator --strict
-TEST_FILE := docker run --rm -v $(TMP_DIR):/repo:ro -e LOG_LEVEL=debug \
+TEST_FILE := docker run --rm -v $(TMP_DIR):/repo:ro -e LOG_LEVEL=debug -e GITHUB_COM_TOKEN \
```
Some questions:
- Where are we generating this token?
- Do we risk exposing it given that we are using `LOG_LEVEL=debug`?

Perhaps we should add a comment line above this one describing why this token if needed, if it's local from the user or from the GHA pipeline, if it has the risk of being exposed or not. I say this, because at a first glance I can't easily say if this has a risk or not.

pjbgf replied:
Added some additional context to the README and Makefile to clarify those points. PTAL

pjbgf committed:
docs: Add GITHUB_COM_TOKEN information for manual testing
For reference, the token is redacted as per debug logs below:

DEBUG: Env config
       'config': {
         'hostRules': [
           {'hostType': 'github', 'matchHost': 'github.com', 'token': '***********'}
         ]
       }
Signed-off-by: Paulo Gomes <paulo.gomes@suse.com>
